x-pack/plugins/reporting/server/lib/content_stream.ts

```diff
@@ -45,55 +171,152 @@ export class ContentStream extends Duplex {
       size: 1,
     };
 
-    try {
-      const response = await this.client.search({ body, index });
-      const hits = response?.body.hits?.hits?.[0] as ReportDocument | undefined;
-      const output = hits?._source.output?.content;
+    const response = await this.client.search<ReportSource>({ body, index });
+    const hits = response?.body.hits?.hits?.[0];
 
-      if (output != null) {
-        this.push(output);
+    this.jobType = hits?._source?.jobtype;
+    this.jobSize = hits?._source?.output?.size;
+
+    return hits?._source?.output?.content;
+  }
+
+  private async readChunk() {
+    const { id, index } = this.document;
+    const body: SearchRequest['body'] = {
+      _source: { includes: ['output.content'] },
+      query: {
+        constant_score: {
+          filter: {
+            bool: {
+              must: [{ term: { parent_id: id } }, { term: { 'output.chunk': this.chunksRead } }],
+            },
+          },
+        },
+      },
+      size: 1,
+    };
+
+    const response = await this.client.search<ChunkSource>({ body, index });
+    const hits = response?.body.hits?.hits?.[0];
+
+    return hits?._source?.output.content;
+  }
+
+  private isRead() {
```

Comment:
The class is about working with the content, and it has it in the name. I would rather keep it that way because it seems a bit redundant to me. It is a private method anyway.